### PR TITLE
Switch to Hugo site generator

### PR DESCRIPTION
**Pull request by wurstsalat3000**
This PR migrates xmpp.org to use Hugo for static site generation.

**Here is what I did:**
* switched to Hugo:
    - migrated templates (processing of members.json, {clients,servers,libraries}.json, xeplist.json)
    - added `prepare_xep_list.py`, which downloads and converts xeplist.xml to xeplist.json
    - adapted `lint-list.py` to directory changes
    - added `announcement.json` for displaying Event announcements on xmpp.org
* created a theme similar to the one used by Pelican for the current deployment, but based on Bootstrap 5.1 (and Fontawesome 5.15.3)
* added a simple user CSS (no compilation necessary)
* added redirects from Pelican URL structure to Hugo URL structure: no .html suffix (aliases in content front matter) + some blog/categories redirects done via /deploy/xsf.conf nginx config file)
* updated Makefiles, Dockerfiles, etc:
   - .travis.yml (adapted to Hugo, passing)
   - Dockerfile (adapted to Hugo and updated to Debian Bullseye, works)
   - fabfile.py (adapted to Hugo, untested)
   - Makefile (adapted to Hugo, works)
   - MakefileDocker (adapted to Hugo, works)
   - Vagrantfile (adapted to Hugo and updated to Ubuntu 21.04, works)

Fixes: #611 , #506 , #468

Obsoletes: #788 , #762 , #743 , #621 , #520 , #508 

For Github to be able to show a diff for changes done to .md files in /content, I first restructured the directory (move/rename) in a separate commit, and then applied changes to the files themselves (mostly concerning frontmatter) in another commit. Viewing the `Migrate content` commit should yield a usable diff when viewed from github.com.

### CODE REVIEW

**Comment by Echolon** [comment]
@mwild1 do you also want to review this?

**Comment by Echolon** [comment]
This is really great work! Thanks for pioneering this approach! @wurstsalat3000 @linkmauve

File: data/libraries.json [hunk 1]
```diff
@@ -28,10 +28,9 @@
     },
     {
         "doap": null,
-        "last_renewed": "2021-07-30T06:24:00",
+        "last_renewed": "2021-01-03T11:24:00",
```
Comment (wurstsalat3000) [inline data/libraries.json:31]: Need to rebase this.

File: data/members.json [hunk 2]
```diff
@@ -68,8 +68,7 @@
         },
 
         {
-            "name": "Daniel Brötzmann",
-            "roles": ["commteam"]
+            "name": "Daniel Brötzmann"
```
Comment (wurstsalat3000) [inline data/members.json:71]: Need to rebase this.

**Review by Zash — APPROVED** [review]
Too big a diff to review in detail. Seems to work in my brief testing.

**Review by mwild1 — APPROVED** [review]

**Pull request merged**